### ISSUE TITLE
Cloud profiles update doesn't work properly

### ISSUE THREAD

**Issue by kayrus** [gardener/garden-setup#105]
**What happened**:

sow uses `kubectl apply -f cloudprofile.yaml`, which does only the update of the yaml file, but doesn't remove the elements, which are not listed in the yaml file anymore.

**What you expected to happen**:

If you remove vulnerable/outdated k8s versions from the cloud profile yaml file, they are not actually removed.

**How to reproduce it (as minimally and precisely as possible)**:

create a cloud profile with the following k8s versions:

* 1.15.0
* 1.15.3
* 1.15.4

then remove the outdated k8s versions and keep only:

* 1.15.4

run the sow, it will produce the following output:

```
manifest for CloudProfile(garden.sapcloud.io/v1beta1): region1
cloudprofile.garden.sapcloud.io/region1 configured
```

but the 1.15.0 and 1.15.3 k8s versions are not removed from the cloud profile.

**Environment**:

- Version of garden-setup (release or commit): 1.15.0

**Comment by Diaphteiros** [comment]
I tried to reproduce this issue using garden-setup 1.16.1, but I failed.

What I tried:
1. deploy garden-setup
2. remove one kubernetes version from the list of kubernetes versions
  2.1 list of kubernetes versions is taken from defaults
  2.2 list of kubernetes versions is taken from acre.yaml (defaults overwritten by `profile` section in iaas entry)
3. deploy cloudprofile again using `sow deploy gardencontent/profiles`

In each case, the deleted kubernetes version has also been removed from the cloudprofile. I even tried it with an active shoot on that version, but it still got removed from the profile.

Since I'm unable to reproduce the issue, I assume it is fixed in the current version of garden-setup. Maybe by the new cloudprofile design (new apigroup) introduced with garden-setup 1.16.0. 
I'll close the issue, feel free to reopen if you can reproduce the problem with garden-setup 1.16+.

**Comment by kayrus** [comment]
probably it was fixed by #108. Considering the text below:

> Due to a bug in the current Gardener version, Openstack shoots that are created with the new manifest (core.gardener.cloud/v1alpha1) have some issues right now. If you are manually creating shoots, use the old manifest to avoid the issues. Since garden-setup uses the new manifest for shooted seeds, you should skip this release and wait for a fix in case you are using a shooted Openstack seed.

Is it already safe to use 1.16 with openstack?
